garak/generators/openai.py

```diff
@@ -169,7 +173,19 @@ def _call_model(
             "stop": self.stop,
         }
 
-        create_args = {k: v for k, v in create_args.items() if v is not None and k not in self.suppressed_params}
+        create_args = {
+            k: v
+            for k, v in create_args.items()
+            if v is not None and k not in self.suppressed_params
+        }
+
+        if self.generator not in (
+            self.client.chat.completions,
+            self.client.completions,
+        ):
+            raise ValueError(
+                "Unsupported model at generation time in generators/openai.py - please add a clause!"
+            )
```

Comment:
It looks like this was just moved so no action to take at this time. Just an inquiry to think on how to approach.

How can this occur?  in theory `_load_client` would raise an error not part of the `backoff` set if it fails to set `self.generator`.

Should we look for a way to enforce validation of this earlier in the run?

Reply:
It should be caught before here - it's def not intended to be mutable after init (ignoring the load/clear client mechanic). I guess `_load_client` is a good place to check, yeah.

Reply:
`_load_client` is a blank method in `OpenAICompatible` which seems a distracting place to put this check; the check is already there in the `OpenAIAPI` class. Moved it to `__init__`, after the first `_load_client`.